torch/csrc/autograd/autograd_not_implemented_fallback.cpp

```diff
@@ -478,6 +478,40 @@ torch::CppFunction autogradNotImplementedFallback() {
       &autogradNotImplementedFallbackImpl>();
 }
 
+struct GenericViewFunc : public ViewFunc {
+  GenericViewFunc(
+      torch::jit::Stack non_tensor_stack,
+      size_t aliased_input_idx_val,
+      c10::OperatorHandle op)
+      : non_tensor_stack_(non_tensor_stack),
+        aliased_input_idx_val_(aliased_input_idx_val),
+        op_(op) {}
+
+  at::Tensor operator()(const at::Tensor& new_base) const override {
+    torch::jit::Stack local_stack = non_tensor_stack_;
+    auto hmmm = c10::IValue(new_base);
+    local_stack.at(aliased_input_idx_val_) = hmmm;
+    op_.callBoxed(local_stack);
+    auto& result = local_stack[local_stack.size() - 1];
+    TORCH_CHECK(
+        result.isTensor(),
+        "ADInplaceOrView fallback view replay did not return a Tensor");
+    return result.toTensor();
+  }
+
+  std::unique_ptr<ViewFunc> clone_and_set(
+      std::optional<std::vector<c10::SymInt>> = std::nullopt,
+      std::optional<std::vector<at::Tensor>> = std::nullopt) const override {
+    return std::make_unique<GenericViewFunc>(
+        non_tensor_stack_, aliased_input_idx_val_, op_);
+  }
```

Comment:
Does this work with compile where the view is passed as an input? Or do we need to figure out how to swap the symints somehow?

Reply:
tbh there was so much logic there I wasn't sure what to do with it.
Do we actually use this ever?

Reply:
It's used for when we fakification views of subclasses. For non-subclasses, we go through the usual as_strided path.

https://github.com/pytorch/pytorch/blob/3c59351c6ea2fc29d346903e28e95c5f4d0ccdbb/torch/_subclasses/meta_utils.py#L1307